tmk_core/protocol/usb_hid/parser.cpp

```diff
@@ -12,11 +12,23 @@ void KBDReportParser::Parse(USBHID *hid, bool is_rpt_id, uint8_t len, uint8_t *b
     }
     dprint("\r\n");
 
-    // ignore Cherry 0101010101010101 bug report
-    // https://geekhack.org/index.php?topic=69169.msg2638223#msg2638223
-    if (buf[1] == 0x01) {
-       dprint("Cherry bug: ignored\r\n");
-       return;
+    // Apple USB Magic keyboard (Model A1644) (buf[1] == 0x01) has mods byte in buf[1] while buf[0] is always 0x01:
+    // input 1: 01 00 00 00 00 00 00 00 00 01
+    // It has 8 key input bytes instead of the usual 6
+    // buf[9] on magic keyboard:
+    // 0x01: eject key (top right next key to F12) pressed
+    // 0x02: fn key (bottom left key) pressed
+    // 0x03: fn and eject key pressed at the same time
+    // we store buf[9] in buf[1] (reserved byte)
+    if (buf[0] == 0x01 && len == 10) {
+        uint8_t t = buf[0];
+        buf[0] = buf[1];
+        buf[1] = buf[9];
```

Comment:
You can assign eject and Fn key as F23 and F24(keys not exist on the keyboard) respectively, for example. Then you can remap the keys in your keymap file.

```
if (buf[9] & 0x01) buf[6] = 0x72 // eject -> F23
if (buf[9] & 0x02) buf[7] = 0x73 // Fn -> F24
```

Reply:
This works perfectly. I am able to then assign F24 to an FN layer.